### PR TITLE
Fix Provider compat CI [3.3, 3.2, 3.1]

### LINKED ISSUE #24246: Provider compat CI is broken because of X25519/X448 having fips=no on master
The two broken jobs here: https://github.com/openssl/openssl/actions/runs/8802704031/job/24159479487 and https://github.com/openssl/openssl/actions/runs/8802704031/job/24159478903 are failing because they run libcrypto/libssl tests from 3.1 and 3.0 branches against the master branch version of the fips provider.

Of course that does not work well because on these branches there are testcases that expect the X25519 and X448 having `fips=yes` property and being advertised by the fips provider among the groups capability.

Furthermore in the provider compat CI we do not run any tests against the 3.2 an 3.3 branches, which would have the same problem.

### PR DESCRIPTION
**Pull request by t8m**
Fixes #24246

### CODE REVIEW

File: .github/workflows/provider-compatibility.yml [hunk 1]
```diff
@@ -109,13 +107,9 @@ jobs:
...
-            name: openssl-3.0,
-            dir: branch-3.0,
-            tgz: branch-3.0.tar.gz,
-          }, {
-            name: openssl-3.1,
-            dir: branch-3.1,
-            tgz: branch-3.1.tar.gz,
+            name: '',
+            dir: HEAD,
+            tgz: HEAD.tar.gz,
```
Comment (paulidale) [inline .github/workflows/provider-compatibility.yml:112]: Doesn't this change lose some of the coverage?
3.0 vs 3.1 e.g.

Comment (t8m) [reply .github/workflows/provider-compatibility.yml:112]: This is just a testing commit to be dropped.

Comment (paulidale) [reply .github/workflows/provider-compatibility.yml:112]: Ah okay, it wasn't obvious.

**Review by paulidale — APPROVED** [review]

**Comment by t8m** [comment]
@paulidale please reconfirm - I've removed all the commits that were to be dropped.

To fully fix the Provider compat CI both #24348 and #24347 must be applied to all the branches as appropriate (some commits do not apply to older branches because the code is not there, they just need to be skipped when cherry picking).

**Review by paulidale — APPROVED** [review]

**Review by tom-cosgrove-arm — APPROVED** [review]
LGTM

**Comment by t8m** [comment]
Merged to the 3.3, 3.2, 3.1 and 3.0 branches (it is already applied on master branch). Thank you for the reviews.

**Pull request closed without merging**